google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,346 @@
+
+import dataclasses
+import re
+import copy
+
+from typing import List, TypeVar
+
+from google.cloud import documentai
+from google.cloud import storage
+
+from google.cloud.documentai_toolbox.wrappers.entity import Entity
+
+from google.cloud.documentai_toolbox.wrappers.document import Document
+
+_OP_TYPE = documentai.Document.Provenance.OperationType
+
+DocumentWithRevisions = TypeVar("DocumentWithRevisions")
+
+
+def _get_base_and_revision_bytes(output_bucket: str, output_prefix: str) -> List[bytes]:
+    r"""Returns a list of shards as bytes.
+
+    If the filepaths are gs://abc/def/gh/{1,2,3}.json, then output_bucket should be "abc",
+    and output_prefix should be "def/gh".
+
+    Args:
+        output_bucket (str):
+            Required. The name of the output_bucket.
+
+        output_prefix (str):
+            Required. The prefix of the folder where files are excluding the bucket name.
+
+    Returns:
+        List[bytes]:
+            A list of shards as bytes.
+
+    """
+    text = ""
+    base_doc = []
+    revisions_doc = []
+
+    storage_client = storage.Client()
+
+    blob_list = storage_client.list_blobs(output_bucket, prefix=output_prefix)
+    pb = documentai.Document.pb()
+    for blob in blob_list:
+        name = blob.name.split("/")[-1]
+
+        if re.search(r"^doc.dp.bp", name) != None:
+            if blob.name.endswith(".dp.bp"):
+                blob_as_bytes = blob.download_as_bytes()
+                text = pb.FromString(blob_as_bytes).text
+        if re.search(r"^pages_*.*", name) != None:
+            if blob.name.endswith(".dp.bp"):
+                blob_as_bytes = blob.download_as_bytes()
+                base_doc.append(blob_as_bytes)
+        elif re.search(r"^rev_*.*", name) != None:
+            if blob.name.endswith(".dp.bp"):
+                blob_as_bytes = blob.download_as_bytes()
+                revisions_doc.append(blob_as_bytes)
+
+    return text, base_doc, revisions_doc
+
+
+def _get_base_docproto(gcs_prefix) -> List[documentai.Document]:
+    """
+    Given a gcs_prefix this funciont will return a list of documentai.Documents
+    from files that follow the pattern pages-.*-to-.*
+    """
+    base_shards = []
+    revision_shards = []
+    text = []
+    match = re.match(r"gs://(.*?)/(.*)", gcs_prefix)
+
+    if match is None:
+        raise ValueError("gcs_prefix does not match accepted format")
+
+    output_bucket, output_prefix = match.groups()
+
+    file_check = re.match(r"(.*[.].*$)", output_prefix)
+
+    if file_check is not None:
+        raise ValueError("gcs_prefix cannot contain file types")
+
+    text, base_bytes, revision_bytes = _get_base_and_revision_bytes(
+        output_bucket, output_prefix
+    )
+    page_pb = documentai.Document.Page.pb()
+    pb = documentai.Document.pb()
+    for byte in base_bytes:
+        doc = documentai.Document()
+        doc.pages = [page_pb.FromString(byte)]
+        doc.text = text
+        base_shards.append(doc)
```

Comment:
Add the fields in the object initialization, although I'm hesitent about adding the pages as an array of one page here, what if the document has multiple pages, how does that work?
```suggestion
        base_shards.append(documentai.Document(
            pages=[page_pb.FromString(byte)],
            text=text
        ))
```